src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/AvaloniaXamlIlLanguage.cs

```diff
@@ -205,67 +207,140 @@ public IXamlCustomAttribute GetCustomAttribute(IXamlProperty property, IXamlType
+                        new[] { size.Width, size.Height });
+                
+                    return true;
+                }
+                catch
+                {
+                    throw new XamlX.XamlLoadException($"Unable to parse \"{text}\" as a size", node);
+                }
             }
             
             if (type.Equals(types.Matrix))
             {
-                var matrix = Matrix.Parse(text);
-
-                result = new AvaloniaXamlIlVectorLikeConstantAstNode(node, types, types.Matrix, types.MatrixFullConstructor,
-                    new[] { matrix.M11, matrix.M12, matrix.M21, matrix.M22, matrix.M31, matrix.M32 });
+                try
+                {
+                    var matrix = Matrix.Parse(text);
+                    
+                    result = new AvaloniaXamlIlVectorLikeConstantAstNode(node, types, types.Matrix, types.MatrixFullConstructor,
+                        new[] { matrix.M11, matrix.M12, matrix.M21, matrix.M22, matrix.M31, matrix.M32 });
                 
-                return true;
+                    return true;
+                }
+                catch
+                {
+                    throw new XamlX.XamlLoadException($"Unable to parse \"{text}\" as a matrix", node);
+                }
             }
             
             if (type.Equals(types.CornerRadius))
             {
-                var cornerRadius = CornerRadius.Parse(text);
-
-                result = new AvaloniaXamlIlVectorLikeConstantAstNode(node, types, types.CornerRadius, types.CornerRadiusFullConstructor,
-                    new[] { cornerRadius.TopLeft, cornerRadius.TopRight, cornerRadius.BottomRight, cornerRadius.BottomLeft });
+                try
+                {
+                    var cornerRadius = CornerRadius.Parse(text);
+            
+                    result = new AvaloniaXamlIlVectorLikeConstantAstNode(node, types, types.CornerRadius, types.CornerRadiusFullConstructor,
+                        new[] { cornerRadius.TopLeft, cornerRadius.TopRight, cornerRadius.BottomRight, cornerRadius.BottomLeft });
                 
+                    return true;
+                }
+                catch
+                {
+                    throw new XamlX.XamlLoadException($"Unable to parse \"{text}\" as a corner radius", node);
+                }
+            }
+            
+            if (type.Equals(types.Color))
+            {
+                if (!Color.TryParse(text, out Color color))
+                {
+                    throw new XamlX.XamlLoadException($"Unable to parse \"{text}\" as a color", node);
+                }
+
+                result = new XamlStaticOrTargetedReturnMethodCallNode(node,
+                    type.GetMethod(
+                        new FindMethodMethodSignature("FromUInt32", type, types.UInt) { IsStatic = true }),
+                    new[] { new XamlConstantNode(node, types.UInt, color.ToUint32()) });
```

Comment:
I think it's better to simplify to constructor call because it looks more clear and faster.

`Color.FromUInt32(16777215U)` -> `new Color(0, 1, 1, 1)`

`FromUInt32` uses constructor call internally.

Reply:
FromUInt32 is way more compact in IL

Reply:
Yeah, my initial thought was that `FromUint32` is more compact in IL since we can avoid push 3 extra arguments. Generally I would see no clear reason why constructor one will be faster too.

Reply:
> Generally I would see no clear reason why constructor one will be faster too.

Because of additional shift and bitwise operations:

```cs
public static Color FromUInt32(uint value)
{
    return new Color(
        (byte)((value >> 24) & 0xff),
        (byte)((value >> 16) & 0xff),
        (byte)((value >> 8) & 0xff),
        (byte)(value & 0xff)
    );
}
```

Reply:
Usually such operations get folded for constants but in this case it seems that we need to force inlining to enable constant folding. This produces very minimal ASM with current IL.

@YohDeadfall Do you know if this might be caused by `FromUInt32` being too big to be inlined by default?

[Sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aAem0MAPLg7582KAE8AfFp0BBLg2tsA4gCEGkSdBZXNezQaMmFjQ0xADMDLDYACYQXJKmDLgYUBxgGAwiHlA0DLkMAN45ecXaPvqGxmaW1MUlOo4wGLgMGAAWMPaSYq3YbqJisTA8DBAAZi3tfVleNbW5pb7+lUGzc2EMwKYYHdYFDADmjWoMAL7Bq7UL5QFVRXMLDU0THWwwUX3ig8Njz1Oed5cdLo/BVAtU5rl1pttgw2HtDhhjmdqAC6mUlmDUXkHo1mm0Oo5YEMPgMuEN0j98X8oDMIQwriCbis6VCtgT4UdTuc6VcMbcLmiGI88ZNnJIOB1IJ8yd9xlT3P8BdigYzluCIayYa58gdOcisZDwploAAKaEdbBoDZsiJW80HO024AASgNBTdxV2AF4GNgNEq5nCfVB/XTio4GD79qGw7lXD7gDG5si6W6ANoAWUarQgUQAkuJJCas21cwWxJIAPJiPixXAsaz7fawXC4XgKGB5uK8Lg9/bOgC6bvWzCQGQgWQYADEoKIAKpdjChUhdyQ9mAmjg99IKbDimCugPuo/FYgAdgYZIA7uOsiaPRCzWznSaTbv9wxzOYGKQUM6GAAZAwdAIKMozOmgD5zE+2wvm+e4Sp+37kEg/5ASBYEQVBtQwQer7vohX4MAAHGhwGgeBkEno+5ovgRHToRRh6xrkzpJrUKZ5MO4Sjre0DTrO+ALjwy6btuDD0cxEKFNRkIXtefFQPesk4bR+EIR0RG/mRGGUdhxS4XB9FIQwKE6UxVEsapz7qR+RGkYB5GYZZVl5IZ8EfoxmHYWxbqcbk+q0DxpAZMexQAusxpQBkHBQLAPAmv+AIySyF5RSwM7zouokgVOeVTnQhV0L5Fz+QwEVGhO/GruuUSJeVFwpRqaVVTSmVCdlK7dmSJq5flRXFexyInEAA===)

Reply:
One of requirements for JIT is the size of IL, it should be less or equal to 20 bytes as far as I remember. That's why it fails to be a candidate for inlining.

Reply:
I'm pleasantly surprised that net 5 is able to detect and fold pure functions.